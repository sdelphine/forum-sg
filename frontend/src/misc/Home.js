import React, { useEffect } from 'react';
import Nav from './Nav';
import { useStoreState, useStoreActions } from 'easy-peasy';
import Logo from'../img/forum.jpg';

export default function Home() {
    const topicsList = useStoreState(states => states.topicsModel.topics.data)
    const fetchTopics = useStoreActions(actions => actions.topicsModel.topics.list);

    useEffect(() => { fetchTopics() }, [])
    const topicsListFiltered = topicsList.filter(topic => topic.slug !== '')
    return (
        <div className="container">
            <img src={Logo} alt='Logo'/>
            <p className='home'>
                Please select a topic
            </p>
            <div className="navbar-home">
                <Nav topicsList={topicsListFiltered}/>
            </div>
        </div>
    )
}