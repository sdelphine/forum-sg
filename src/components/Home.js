import React, { Component } from 'react';
import Header from '../components/Header';
import MessageList from '../components/MessageList';
import { Link } from 'react-router-dom';

class Home extends Component {
    state = {
        messageList: [
            { id: '1', topic: 'legal', message: 'First message', from: 'Toto', at: 'Today at 2:00 PM' },
            { id: '2', topic: 'legal', message: 'Second message', from: 'Titi', at: 'Today at 4:00 PM' },
            { id: '3', topic: 'legal', message: 'Third message', from: 'Toto', at: 'Tomorrow at 10:00 AM' }
        ]
    }

    render() {
        return (
            <div className='home'>
                <Header />
                <MessageList messageList={this.state.messageList}/>
                <Link to='/newMessage'>ðŸ“§ Add a new message</Link>
            </div>
        );
    }
}

export default Home;