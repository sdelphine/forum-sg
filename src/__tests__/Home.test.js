import { shallow, mount, render } from 'enzyme';
import Home from '../components/Home';
import React from 'react';
import { MemoryRouter as Router, Link } from 'react-router';
import toJSON from 'enzyme-to-json';

const wrapper = mount(
    <Router>
        <Home />
    </Router>
)

describe('<Home />', () => {
    it('render all home components', () => {
        const part = wrapper.find('.home')
        expect(part.children()).toHaveLength(3)
    });
    
    it('render header component', () => {
        const part = wrapper.find('Header')
        expect(toJSON(part)).toMatchSnapshot()
    });

    it('render message component', () => {
        const part = wrapper.find('MessageList')
        expect(toJSON(part)).toMatchSnapshot()
    })

    it('includes link to add message page', () => {                                       
        const part = wrapper.find('Link')
        expect(part.props().to).toBe('/newMessage');
    });
});