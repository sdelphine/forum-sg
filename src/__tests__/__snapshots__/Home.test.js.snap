// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> render header component 1`] = `
<Header>
  <div>
    <h1>
      Header
    </h1>
  </div>
</Header>
`;

exports[`<Home /> render message component 1`] = `
<MessageList>
  <Message
    key="1"
    message={
      Object {
        "at": "Today at 2:00 PM",
        "from": "Toto",
        "id": "1",
        "message": "First message",
        "topic": "legal",
      }
    }
  >
    <div
      className="message-box"
    >
      <p
        className="message-at"
      >
        At 
        Today at 2:00 PM
      </p>
      <p
        className="message-from"
      >
        From 
        Toto
      </p>
      <p
        className="message"
      >
        First message
      </p>
      <br />
    </div>
  </Message>
  <Message
    key="2"
    message={
      Object {
        "at": "Today at 4:00 PM",
        "from": "Titi",
        "id": "2",
        "message": "Second message",
        "topic": "legal",
      }
    }
  >
    <div
      className="message-box"
    >
      <p
        className="message-at"
      >
        At 
        Today at 4:00 PM
      </p>
      <p
        className="message-from"
      >
        From 
        Titi
      </p>
      <p
        className="message"
      >
        Second message
      </p>
      <br />
    </div>
  </Message>
  <Message
    key="3"
    message={
      Object {
        "at": "Tomorrow at 10:00 AM",
        "from": "Toto",
        "id": "3",
        "message": "Third message",
        "topic": "legal",
      }
    }
  >
    <div
      className="message-box"
    >
      <p
        className="message-at"
      >
        At 
        Tomorrow at 10:00 AM
      </p>
      <p
        className="message-from"
      >
        From 
        Toto
      </p>
      <p
        className="message"
      >
        Third message
      </p>
      <br />
    </div>
  </Message>
</MessageList>
`;
